### PR TITLE
docs: add reclient docs, remove goma docs

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Documents reclient in the same way Goma was, removes goma docs and points to reclient

Notes: none

### CODE REVIEW

File: docs/development/reclient.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+If you have an existing config, you can just set `"reclient": "remote_exec"`
+in your config file.
+
+## Building with Reclient
+
+When you are using Reclient you can run `autoninja` with a substantially higher `j`
+value than would normally be supported by your machine.
+
+Please do not set a value higher than **200**. We monitor RBE system usage, and users
+found to be abusing it with unreasonable concurrency will be de-activated.
```
Comment (ckerr) [inline docs/development/reclient.md:26]:
```suggestion
Please do not set a value higher than **200**. The RBE system is monitored.
Users found to be abusing it with unreasonable concurrency will be deactivated.
```

File: docs/development/reclient.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+to Electron Maintainers.  If you want access please head to `#access-requests` in
+Slack and ping `@infra-wg` to ask for access.  Please be aware that being a
+maintainer does not _automatically_ grant access and access is determined on a
+case by case basis.
+
+## Support
+
+We do not provide support for usage of Reclient and any issues raised asking for help / having
+issues will _probably_ be closed without much reason, we do not have the capacity to handle
+that kind of support.
```
Comment (ckerr) [inline docs/development/reclient.md:46]:
```suggestion
We do not provide support for usage of Reclient. Issues raised asking for help / having
issues will _probably_ be closed without much reason. We do not have the capacity to handle
that kind of support.
```

File: docs/development/reclient.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+autoninja -C out/Testing electron -j 200
+```
+
+If you're using `build-tools`, appropriate `-j` values will automatically be used for you.
+
+## Access
+
+For security and cost reasons, access to Electron's RBE backend is currently restricted
+to Electron Maintainers.  If you want access please head to `#access-requests` in
+Slack and ping `@infra-wg` to ask for access.  Please be aware that being a
```
Comment (ckerr) [inline docs/development/reclient.md:38]:
```suggestion
to Electron Maintainers.  If you want access, please head to `#access-requests` in
Slack and ping `@infra-wg` to ask for it.  Please be aware that being a
```

File: docs/development/reclient.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+maintainer does not _automatically_ grant access and access is determined on a
+case by case basis.
```
Comment (ckerr) [inline docs/development/reclient.md:40]:
```suggestion
maintainer does not _automatically_ grant access. Access is determined on a
case-by-case basis.
```

File: docs/development/reclient.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+Currently the only supported way to use Reclient is to use our [Build Tools](https://github.com/electron/build-tools).
+Reclient configuration is automatically included when you set up `build-tools`.
+
+If you have an existing config, you can just set `"reclient": "remote_exec"`
+in your config file.
+
+## Building with Reclient
+
+When you are using Reclient you can run `autoninja` with a substantially higher `j`
```
Comment (ckerr) [inline docs/development/reclient.md:22]:
```suggestion
When you are using Reclient, you can run `autoninja` with a substantially higher `j`
```

File: docs/development/reclient.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Reclient
+
+> Reclient integrates with an existing build system to enable remote execution and caching of build actions.
+
+Electron has a deployment of a [reclient](https://github.com/bazelbuild/reclient)
+compatible RBE Backend that we make available to all Electron Maintainers.
```
Comment (ckerr) [inline docs/development/reclient.md:6]:
```suggestion
compatible RBE Backend that is available to all Electron Maintainers.
```

Also, "available to all Electron maintainers" here is inconsistent with the "being a maintainer doesn't automatically get you access" text below

File: docs/development/reclient.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+See the [Access](#access) section below for details on authentication. Non maintainer
+will have access to a `cache-only` CAS backend, you will still need to sign in
+but will receive a `Cache Only` token.  Requests with Cache Only tokens will not hit
+our cluster, but will read from our cache and should result in significantly faster build times.
```
Comment (ckerr) [inline docs/development/reclient.md:10]:
```suggestion
See the [Access](#access) section below for details on authentication. Non-maintainers
will not have access to the cluster, but can sign in to receive a `Cache Only` token
that gives access to the cache-only CAS backend. Using this should result in
significantly faster build times .

**Review by ckerr — APPROVED** [review]
LGTM.

I had some copyediting suggestions, e.g. grammar and minor phrasing, but the substance is fine

**Review by VerteDinde — APPROVED** [review]

**Pull request merged**